src/plugins/offline_editing/offline_editing_plugin_gui.cpp

```diff
@@ -30,33 +30,67 @@
 #include <QFileDialog>
 #include <QMessageBox>
 
-
 QgsSelectLayerTreeModel::QgsSelectLayerTreeModel( QgsLayerTree *rootNode, QObject *parent )
   : QgsLayerTreeModel( rootNode, parent )
 {
   setFlag( QgsLayerTreeModel::ShowLegend, false );
   setFlag( QgsLayerTreeModel::AllowNodeChangeVisibility, true );
 }
 
+int QgsSelectLayerTreeModel::columnCount( const QModelIndex &parent ) const
+{
+  Q_UNUSED( parent );
+  return QgsLayerTreeModel::columnCount( parent ) + 1;
+}
+
+
 QVariant QgsSelectLayerTreeModel::data( const QModelIndex &index, int role ) const
 {
-  if ( role == Qt::CheckStateRole )
+  QgsLayerTreeNode *node = index2node( index );
+  if ( index.column() == 0 )
   {
-    QgsLayerTreeNode *node = index2node( index );
-    if ( QgsLayerTree::isLayer( node ) )
+    if ( role == Qt::CheckStateRole )
     {
-      QgsLayerTreeLayer *nodeLayer = QgsLayerTree::toLayer( node );
-      return nodeLayer->isVisible();
+      if ( QgsLayerTree::isLayer( node ) )
+      {
+        QgsLayerTreeLayer *nodeLayer = QgsLayerTree::toLayer( node );
+        return nodeLayer->isVisible();
+      }
+      else if ( QgsLayerTree::isGroup( node ) )
+      {
+        QgsLayerTreeGroup *nodeGroup = QgsLayerTree::toGroup( node );
+        return nodeGroup->isVisible();
+      }
+      else
+      {
+        return QVariant();
+      }
     }
-    else if ( QgsLayerTree::isGroup( node ) )
-    {
-      QgsLayerTreeGroup *nodeGroup = QgsLayerTree::toGroup( node );
-      return nodeGroup->isVisible();
-    }
-    else
+  }
+  else
+  {
+    if ( QgsLayerTree::isLayer( node ) && index.column() > 0 )
     {
-      return QVariant();
+      QgsLayerTreeLayer *nodeLayer = QgsLayerTree::toLayer( node );
+      if ( nodeLayer->layer()->dataProvider()->name() == QStringLiteral( "WFS" ) )
+      {
+        switch ( role )
+        {
+          case Qt::ToolTipRole:
+            return tr( "The source of this layer is a <b>WFS</b> server.<br>"
+                       "Some WFS layers are not suitable for offline<br>"
+                       "editing due to unrealiable/missing primary<br>"
```

Comment:
unreliable

Reply:
:+1:

Reply:
unstable?

Reply:
mmm it depends if you refer to the cause or to the effect, to the remote back-end or to the generated XML. The cause is usually that the back-end data source has a missing PK (like in shapefiles), the effect is that the generated feature id (let's call it PK for simplicity) in the XML is unreliable *because*  it is unstable. 
How about `unreliable (unstable) primary key` ?
I didn't want to write a wikipedia page though :wink:

Reply:
I just wanted to make it as easy for an admin to understand for "your system administrator", so the term "unstable primary key" seemed to be most precise to me ;)